### PR TITLE
Convert Tubular to Python 3

### PR DESCRIPTION
**Pull request by cpennington**
Until edx-gomatic is converted to call the tubular scripts using the shebang, tubular should stay python2/3 compatible.

### CODE REVIEW

File: tubular/tests/test_asgard.py [hunk 1]
```diff
@@ -29,336 +29,328 @@
...
+            "status": "deleted",
+            "launchingSuspended": False,
+            "desiredCapacity": 4,
+            "minSize": 4
+        },
+        "clusterName": "app_cluster"
+    }
+
+
+def deleted_asg_not_in_progress(name):
```
Comment (macdiesel) [inline tubular/tests/test_asgard.py:196]: nice change. 👍

File: tubular/tests/test_asgard.py [hunk 2]
```diff
@@ -381,228 +373,188 @@ def test_clusters_for_asgs(self):
...
 
-    @httpretty.activate
-    def test_new_asg(self):
+    def test_new_asg(self, req_mock):
         task_url = "http://some.host/task/1234.json"
         cluster = "loadtest-edx-edxapp"
         ami_id = "ami-abc1234"
 
-        def post_callback(request, uri, headers):  # pylint: disable=unused-argument
+        def post_callback(request, context):  # pylint: disable=unused-argument
```
Comment (macdiesel) [inline tubular/tests/test_asgard.py:650]: I think you can remove the pylint annotation.

File: tubular/tests/test_asgard.py [hunk 3]
```diff
@@ -649,216 +601,202 @@ def test_new_asg_failure(self,
...
             Callback method for POST.
             """
             self.assertEqual('POST', request.method)
             expected_request_body = {"name": [cluster], "imageId": [ami_id]}
             expected_querystring = {"asgardApiToken": ['dummy-token']}
 
-            self.assertEqual(expected_request_body, request.parsed_body)
-            self.assertEqual(expected_querystring, request.querystring)
-            return create_response_code, create_response_headers, create_response_body
+            parsed_body = six.moves.urllib.parse.parse_qs(request.text)
```
Comment (macdiesel) [inline tubular/tests/test_asgard.py]: from six.moves import urllib to shorten this up?

File: tox.ini [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+[testenv]
+envdir =
+    py27: {toxworkdir}/py27
+    py34: {toxworkdir}/py34
+    py35: {toxworkdir}/py35
+deps=-rrequirements/test_requirements.txt
+commands=
+    test: pytest {posargs:-n auto}
+	quality: pep8 --config=.pep8 tubular scripts admin
```
Comment (feanil) [inline tox.ini]: Is this spacing correct or relevant?  I don't see anything about it in the tox docs.

File: tox.ini [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+	quality-py27: pylint --py3k tubular scripts admin
```
Comment (feanil) [inline tox.ini]: Am I correct in assuming this will only get run on python 2.7?  Does the order in the naming of the key not matter?

**Review by feanil — COMMENTED** [review]
Generally looks good, just a couple of questions around tox.

File: tox.ini (hunk 4 above)
Comment (cpennington) [reply tox.ini]: Whoah, strange.

File: tox.ini (hunk 5 above)
Comment (cpennington) [reply tox.ini]: The order doesn't matter (it just does set comparison on the individual factors). But, having answered this question multiple times, I'm just going to use a consistent order.

File: tubular/confluence_api.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
+"""Stuff"""
```
Comment (doctoryes) [inline tubular/confluence_api.py]: Oops?

**Review by doctoryes — APPROVED** [review]
Once confluence_api.py is removed.

File: tubular/confluence_api.py (hunk 6 above)
Comment (cpennington) [reply tubular/confluence_api.py]: Drat. Didn't mean to check that file in.

**Pull request merged**